testing/zkevm-lib/src/witness.rs

```diff
@@ -0,0 +1,69 @@
+use crate::SP1Input;
+use reth_interfaces::provider::ProviderError;
+use reth_primitives::B256;
+use revm::DatabaseRef;
+use revm_primitives::{AccountInfo, Address, Bytecode, HashMap, U256};
+
+pub struct WitnessDb {
+    pub address_to_account_info: HashMap<Address, AccountInfo>,
+    pub address_to_storage: HashMap<Address, HashMap<U256, U256>>,
+    pub block_hashes: HashMap<U256, B256>,
+    pub(crate) state_root: B256,
+}
+
+impl WitnessDb {
+    pub fn new(sp1_input: SP1Input) -> Self {
+        let state_root: B256 = sp1_input.prev_block.state_root.into();
+        let mut address_to_account_info = HashMap::new();
+        let mut address_to_storage = HashMap::new();
+
+        for (address, proof) in sp1_input.address_to_proof {
+            println!("Verifying account proof for address {address:?}...");
+            proof.verify(state_root).expect("account proof verification failed");
```

Comment:
I think this is missing validation that `address == proof.address`, since in `proof.verify()` it uses its own `proof.address` field for the verification.